app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala

```diff
@@ -55,4 +55,15 @@ trait CommonFormValidators {
   val isValidTelephoneNumber: String => Boolean = (value: String) =>
     value.isEmpty || isMatchingPattern(value, phoneNumberRegexPattern)
 
+  val addressInputRegexPattern: Pattern = Pattern.compile("^[A-Za-z0-9 \\-,.&']{1,35}$")
+
+  val isValidAddressInput: String => Boolean = (value: String) =>
+    value.isEmpty || isMatchingPattern(value, addressInputRegexPattern)
+
+  val postcodeRegexPattern: Pattern =
+    Pattern.compile("^[A-Z]{1,2}[0-9][0-9A-Z]?\\s?[0-9][A-Z]{2}|BFPO\\s?[0-9]{1,10}$")
```

Comment:
Just curious, where did you get this regex from?

Reply:
EIS regex pattern from one of their API resources.
Keep in mind that this is for the legacy/current API resources, we will have to use new API resources, so this means we will have to use potentially a new regex. We haven't received the API docs yet, so we are going with what they expect at the moment.

Reply:
👍   Makes sense.